src/lib.rs

```diff
@@ -481,16 +487,21 @@ impl LineParser {
 ///
 /// For basic usage, see the crate-level documentation.
 /// Note that once `next` returns an error, the result of further calls is undefined.
-pub struct PropertiesIter<R: Read> {
-  lines: LogicalLines<NaturalLines<R>>,
+pub struct PropertiesIter<'a, R: Read> {
+  lines: LogicalLines<NaturalLines<'a, R>>,
   parser: LineParser,
 }
 
-impl<R: Read> PropertiesIter<R> {
+impl<'a, R: Read> PropertiesIter<'a, R> {
   /// Parses properties from the given `Read` stream.
   pub fn new(input: R) -> Self {
+    Self::new_with_encoding(input, ISO_8859_1)
+  }
+
+  /// Parses properties from the given `Read` stream in the given encoding
+  pub fn new_with_encoding(input: R, encoding: &'a Encoding) -> Self {
```

Comment:
This should have a warning on it that using an encoding other than ISO 8859-1 is not strictly compliant with the  Java properties file spec, and that most users should call `new` instead.